braindecode/regressor.py

```diff
@@ -70,155 +63,13 @@ def __init__(self, *args, cropped=False, callbacks=None,
         iterator = super().get_iterator(dataset, training=training)
         if drop_index:
             return ThrowAwayIndexLoader(self, iterator, is_regression=True)
         else:
             return iterator
 
-    def on_batch_end(self, net, *batch, training=False, **kwargs):
-        # If training is false, assume that our loader has indices for this
-        # batch
-        if not training:
-            epoch_cbs = []
-            for name, cb in self.callbacks_:
-                if isinstance(cb, (CroppedTrialEpochScoring, CroppedTimeSeriesEpochScoring)) and (
-                        hasattr(cb, 'window_inds_')) and (not cb.on_train):
-                    epoch_cbs.append(cb)
-            # for trialwise decoding stuffs it might also be we don't have
-            # cropped loader, so no indices there
-            if len(epoch_cbs) > 0:
-                assert self._last_window_inds_ is not None
-                for cb in epoch_cbs:
-                    cb.window_inds_.append(self._last_window_inds_)
-                self._last_window_inds_ = None
-
-    def predict_with_window_inds_and_ys(self, dataset):
-        self.module.eval()
-        preds = []
-        i_window_in_trials = []
-        i_window_stops = []
-        window_ys = []
-        for X, y, i in self.get_iterator(dataset, drop_index=False):
-            i_window_in_trials.append(i[0].cpu().numpy())
-            i_window_stops.append(i[2].cpu().numpy())
-            with torch.no_grad():
-                preds.append(to_numpy(self.module.forward(X.to(self.device))))
-            window_ys.append(y.cpu().numpy())
-        preds = np.concatenate(preds)
-        i_window_in_trials = np.concatenate(i_window_in_trials)
-        i_window_stops = np.concatenate(i_window_stops)
-        window_ys = np.concatenate(window_ys)
-        return dict(
-            preds=preds, i_window_in_trials=i_window_in_trials,
-            i_window_stops=i_window_stops, window_ys=window_ys)
-
-    # Removes default EpochScoring callback computing 'accuracy' to work properly
-    # with cropped decoding.
-    @property
-    def _default_callbacks(self):
-        return [
-            ("epoch_timer", EpochTimer()),
-            (
-                "train_loss",
-                BatchScoring(
-                    train_loss_score,
-                    name="train_loss",
-                    on_train=True,
-                    target_extractor=noop,
-                ),
-            ),
-            (
-                "valid_loss",
-                BatchScoring(
-                    valid_loss_score, name="valid_loss", target_extractor=noop,
-                ),
-            ),
-            ("print_log", PrintLog()),
-        ]
-
     # Method added to fix `predict_proba` behavior to return proper values
     # in cropped mode even if regressor does not return probabilities.
     # We implement `predict_proba` because it exists in skorch.NeuralNetRegressor.
```

Comment:
What does this comment refer to?

Reply:
doesn't make sense

Reply:
this comment refers to the existence of `predict_proba` for Regressor -> it doesn't make much sense to have `predict_proba` method for regressors as there is no probability returned. We actually implemented the `predict_proba` because it was implemented in skorch also for regressor and in the case we don't implement it here, then someone could use `predict_proba` not adjusted to cropped decoding.